HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,244 @@
+    std::cout << "\nConverting Axis-Angle to Rotation Vector:" << std::endl;
+    if(!axisAngle.axis().isUnitary())
+    {
+        std::cout << "Note! Axis in Axis-Angle pair is not unitary" << std::endl;
+    }
+    robotRepresentations.rotationVector = axisAngle.angle() * axisAngle.axis();
+    std::cout << robotRepresentations.rotationVector << std::endl;
+
+    // Rotation Matrix to Quaternion
+    std::cout << "\nConverting Rotation Matrix to Quaternion:" << std::endl;
+    const Eigen::Quaterniond quaternion(transformationMatrix.linear());
+    robotRepresentations.quaternion = quaternion;
+    std::cout << robotRepresentations.quaternion.coeffs() << std::endl;
+
+    // Rotation Matrix to Roll-Pitch-Yaw
+    for(int i = 0; i < nofRotationConventions; i++)
+    {
+        const RotationConvention convention{ static_cast<RotationConvention>(i) };
+        std::cout << "\nConverting Rotation Matrix to Roll-Pitch-Yaw angles (" << toString(convention)
+                  << "):" << std::endl;
+        robotRepresentations.rotations[i] = { static_cast<RotationConvention>(i),
+                                              rotmat2rpy(transformationMatrix.linear(), convention) };
+        std::cout << robotRepresentations.rotations[i].rollPitchYaw << std::endl;
+    }
+
+    return robotRepresentations;
+}
+
+Eigen::Affine3d transformFromRobotToZivid(const Eigen::Affine3d &transformationMatrix,
+                                          const Representations &representations)
+{
+    // Roll, Pitch and Yaw to Rotation Matrix
+    for(const auto &rotation : representations.rotations)
+    {
+        std::cout << "\nConverting Roll-Pitch-Yaw angles (" << toString(rotation.convention)
+                  << ") to Rotation Matrix:" << std::endl;
+        const Eigen::Matrix3d rotationMatrixFromRollPitchYaw = rpy2rotmat(rotation.rollPitchYaw, rotation.convention);
+        std::cout << rotationMatrixFromRollPitchYaw << std::endl;
+    }
+
+    // Rotation Vector to Axis-angle
+    std::cout << "\nConverting Rotation Vector to Axis-Angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle(representations.rotationVector.norm(),
+                                      representations.rotationVector.normalized());
+    std::cout << "Axis:\n" << axisAngle.axis() << std::endl;
+    std::cout << "Angle:\n" << axisAngle.angle() << std::endl;
+
+    // Axis-Angle to Quaternion
+    std::cout << "\nConverting Axis-Angle to Quaternion:" << std::endl;
+    const Eigen::Quaterniond quaternion(axisAngle);
+    std::cout << quaternion.coeffs() << std::endl;
+
+    // Quaternion to Rotation Matrix
+    std::cout << "\nConverting Quaternion to Rotation Matrix:" << std::endl;
+    auto transformationMatrixFromQuaternion = Eigen::Affine3d(transformationMatrix);
+    transformationMatrixFromQuaternion.linear() = quaternion.toRotationMatrix();
+    std::cout << transformationMatrixFromQuaternion.linear() << std::endl;
+
+    return transformationMatrixFromQuaternion;
```

Comment:
I would just call this one transformationMatrix

Reply:
Only if we end up changing input.

Reply:
Even if not, I would get rid of 'Quaternion' and rename somehow. transformationMatrixFromQuaternionAndTranslationVector would be correct but it is not nice.